server/sonar-server/src/main/java/org/sonar/server/ui/ws/ComponentConfigurationPages.java

```diff
@@ -0,0 +1,156 @@
+    Locale locale = userSession.locale();
+    String componentKey = encodeComponentKey(component);
+
+    List<ConfigPage> configPages = Lists.newArrayList();
+
+    configPages.add(new ConfigPage(
+      isAdmin && componentTypeHasProperty(component, PROPERTY_CONFIGURABLE),
+      String.format("/project/settings?id=%s", componentKey),
+      i18n.message(locale, "project_settings.page", null)));
+
+    configPages.add(new ConfigPage(
+      isProject,
+      String.format("/project/profile?id=%s", componentKey),
+      i18n.message(locale, "project_quality_profiles.page", null)));
+
+    configPages.add(new ConfigPage(
+      isProject,
+      String.format("/project/qualitygate?id=%s", componentKey),
+      i18n.message(locale, "project_quality_gate.page", null)));
+
+    configPages.add(new ConfigPage(
+      isAdmin,
+      String.format("/manual_measures/index?id=%s", componentKey),
+      i18n.message(locale, "manual_measures.page", null)));
+
+    configPages.add(new ConfigPage(
+      isAdmin && isProject,
+      String.format("/action_plans/index?id=%s", componentKey),
+      i18n.message(locale, "action_plans.page", null)));
+
+    configPages.add(new ConfigPage(
+      isAdmin && isProject,
+      String.format("/project/links?id=%s", componentKey),
+      i18n.message(locale, "action_plans.page", null)));
+
+    configPages.add(new ConfigPage(
+      componentTypeHasProperty(component, PROPERTY_HAS_ROLE_POLICY),
+      String.format("/project_roles/index?id=%s", componentKey),
+      i18n.message(locale, "permissions.page", null)));
+
+    configPages.add(new ConfigPage(
+      componentTypeHasProperty(component, PROPERTY_MODIFIABLE_HISTORY),
+      String.format("/project/history?id=%s", componentKey),
+      i18n.message(locale, "project_history.page", null)));
+
+    configPages.add(new ConfigPage(
+      componentTypeHasProperty(component, PROPERTY_UPDATABLE_KEY),
+      String.format("/project/key?id=%s", componentKey),
+      i18n.message(locale, "update_key.page", null)));
+
+    configPages.add(new ConfigPage(
+      componentTypeHasProperty(component, PROPERTY_DELETABLE),
+      String.format("/project/deletion?id=%s", componentKey),
+      i18n.message(locale, "deletion.page", null)));
+
+    return configPages;
+  }
+
+  static String encodeComponentKey(ComponentDto component) {
+    String componentKey = component.getKey();
+    try {
+      componentKey = URLEncoder.encode(componentKey, Charsets.UTF_8.name());
+    } catch (UnsupportedEncodingException unknownEncoding) {
+      throw new IllegalStateException(unknownEncoding);
+    }
+    return componentKey;
+  }
+
+  boolean componentTypeHasProperty(ComponentDto component, String resourceTypeProperty) {
+    ResourceType resourceType = resourceTypes.get(component.qualifier());
+    if (resourceType != null) {
+      return resourceType.getBooleanProperty(resourceTypeProperty);
+    }
+    return false;
+  }
+
+  static class ConfigPage {
```

Comment:
Note that his class and its methods can be private instead of package protected.